synapse/event_auth.py

```diff
@@ -455,8 +455,11 @@ def _check_create(event: "EventBase") -> None:
             "room appears to have unsupported version %s" % (room_version_prop,),
         )
 
-    # 1.4 If content has no creator field, reject.
-    if EventContentFields.ROOM_CREATOR not in event.content:
+    # 1.4 If content has no creator field, reject if the room version requires it.
+    if (
+        not event.room_version.msc2175_create_events
+        and EventContentFields.ROOM_CREATOR not in event.content
```

Comment:
Can we leave a comment near the definition of ROOM_CREATOR to say that it's been deprecated by this MSC?

Reply:
Arg, I meant to do that, thank you for the reminder.